src/uct/ib/base/ib_iface.c

```diff
@@ -813,6 +840,25 @@ static ucs_status_t uct_ib_iface_set_moderation(struct ibv_cq *cq,
     return UCS_OK;
 }
 
+static void uct_ib_iface_set_num_paths(uct_ib_iface_t *iface,
+                                       const uct_ib_iface_config_t *config)
+{
+    uct_ib_device_t *dev = uct_ib_iface_device(iface);
+
+    if (config->num_paths == UCS_ULUNITS_AUTO) {
+        if (uct_ib_iface_is_roce(iface)) {
+            /* RoCE - number of paths is RoCE LAG level */
+            iface->num_paths = uct_ib_device_get_roce_lag_level(dev);
+        } else {
+            /* IB - number of paths is LMC level */
+            ucs_assert(iface->path_bits_count > 0);
+            iface->num_paths = iface->path_bits_count;
+        }
+    } else {
+        iface->num_paths = config->num_paths;
```

Comment:
do we need to check for wrong values?

Reply:
we use modulo so if num_paths>path_bits_count it wraps around